cpp/src/parquet/arrow/reader.cc

```diff
@@ -803,6 +809,9 @@ Status FileReaderImpl::ReadRowGroups(const std::vector<int>& row_groups,
     return Status::Invalid("Invalid column index");
   }
 
+  // PARQUET-1698/PARQUET-1820: pre-buffer row groups/column chunks if enabled
+  parquet_reader()->PreBuffer(row_groups, indices);
```

Comment:
Per comments below, it seems slightly odd that this would be a no-op sometimes. It seems like it would be better to have

```
if (properties_.prebuffering_enabled()) {
  parquet_reader()->PreBuffer(...);
}
```